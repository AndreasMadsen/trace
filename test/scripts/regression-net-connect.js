'use strict';

const net = require('net');

//
// test an issue where there are multiply internal async events
//

const socket = net.connect(24075); // connection refused
socket.once('error', function (error) {
  throw error;
});
