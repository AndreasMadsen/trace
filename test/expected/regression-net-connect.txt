/scripts/regression-net-connect.js:11
  throw error;
  ^

Error: connect ECONNREFUSED 127.0.0.1:24075
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:r:c)
    at new TickObject (internal/process/next_tick.js:r:c)
    at process.nextTick (internal/process/next_tick.js:r:c)
    at _destroy (internal/streams/destroy.js:r:c)
    at Socket._destroy (net.js:r:c)
    at Socket.destroy (internal/streams/destroy.js:r:c)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:r:c)
    at internalConnect (net.js:r:c)
    at GetAddrInfoReqWrap.emitLookup [as callback] (net.js:r:c)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:r:c)
    at Socket.connect (net.js:r:c)
    at Object.connect (net.js:r:c)
    at Object.<anonymous> (/scripts/regression-net-connect.js:r:c)
    at Module._compile (internal/modules/cjs/loader.js:r:c)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:r:c)
    at Module.load (internal/modules/cjs/loader.js:r:c)
    at tryModuleLoad (internal/modules/cjs/loader.js:r:c)
    at Function.Module._load (internal/modules/cjs/loader.js:r:c)
    at Function.Module.runMain (internal/modules/cjs/loader.js:r:c)
    at startup (internal/bootstrap/node.js:r:c)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:r:c)
